app/main/views.py

```diff
@@ -35,21 +35,45 @@ def new_post():
 @main.route('/staff-directory', methods=['GET', 'POST'])
 def staff_directory():
     form = Staff_Directory_Search_Form()
+    page = flask_request.args.get('page', 1, type=int)
 
     if form.search.data is "":
-        users = Users.query.order_by(Users.last_name)
+        users = Users.query.order_by(Users.last_name).paginate(page=page, per_page=10)
+
     elif form.filters.data == 'First Name':
-        users = Users.query.filter(Users.first_name.ilike('%' + form.search.data + '%'))
+        return redirect('/search/'+form.filters.data+'/'+form.search.data)
     elif form.filters.data == 'Last Name':
-        users = Users.query.filter(Users.last_name.ilike('%' + form.search.data + '%'))
+        return redirect('/search/' + form.filters.data + '/' + form.search.data)
     elif form.filters.data == 'Division':
-        users = Users.query.filter(Users.division.ilike('%' + form.search.data + '%'))
+        return redirect('/search/' + form.filters.data + '/' + form.search.data)
     elif form.filters.data == 'Title':
-        users = Users.query.filter(Users.title.ilike('%' + form.search.data + '%'))
+        return redirect('/search/' + form.filters.data + '/' + form.search.data)
     else:
-        users = Users.query.order_by(Users.last_name)
+        users = Users.query.order_by(Users.last_name).paginate(page=page, per_page=10)
+
+    return render_template('staff_directory.html', users=users, form=form, page=page)
+
+@main.route('/search/<string:filter>/<string:input>', methods=['GET', 'POST'])
+def show_result(filter,input):
+    form = Staff_Directory_Search_Form()
```

Comment:
You should put a docstring explaining what this function does and what it returns.